src/components/main/filmsList/FilmsList.tsx

```diff
@@ -1,63 +1,98 @@
+    const categories = useSelector((state: ActionTypeInterface) => state.pageCounter.categories)
+    const sortType = useSelector((state: ActionTypeInterface) => state.sortType.type)
+    const isLoaded = useSelector((state: ActionTypeInterface) => state.movieLoader.isLoaded)
+
+    const search = useSelector((state: ActionTypeInterface) => state.mainBannerAndSearch.search)
 
     useEffect(() => {
-        if (pageCounter > 1) {
-            getFilmlist(categories, pageCounter)
-                .then(res => setFilmArr(() => ([...filmArr, ...res.results])))
+        if (search.length > 0) {
+            // const handle: ReturnType<typeof setTimeout> = setTimeout(() => {
+            getSearchResult(search)
+                .then(res => {
+                    if (res.results.length > 0) {
+                        setFilmArr(res.results)
+                        dispatch(changeLoading(true))
+                    } else {
+                        console.log('e')
+                    }
+
+                })
+            // }, 1000)
+
+            // return () => clearTimeout(handle)
         } else {
-            getFilmlist(categories, pageCounter)
-                .then(res => setFilmArr(res.results))
+            if (pageCounter > 1) {
+                getFilmlist(categories, pageCounter)
+                    .then(res => {
+                        setFilmArr(() => ([...filmArr, ...res.results]))
+                        dispatch(changeLoading(true))
+                    })
+            } else {
+                getFilmlist(categories, pageCounter)
+                    .then(res => {
+                        setFilmArr(res.results)
+                        dispatch(changeLoading(true))
+                    })
+            }
         }
-    }, [categories, pageCounter])
+
+    }, [categories, pageCounter, search])
 
     useEffect(() => {
         getGenres()
             .then(res => setGenres(res.genres))
     }, [])
 
-
+    console.log(filmArr)
     return (
         <>
-            <div className={sortType ? 'filmListRow' : 'filmListColumn'}>
-                {
-                    filmArr.map((el: FilmInterface, i: number) => {
-                        return (
-                            <FilmItem
-                                sortType={sortType}
-                                key={uniqid()}
-                                title={el.title}
-                                image={el.poster_path}
-                                rate={el.vote_average}
-                                overview={el.overview}
-                                filmGenres={el.genre_ids}
-                                genres={genres}
-                            />
-                        )
-                    })
-                }
-            </div>
+            {isLoaded ?
+                <div className={sortType ? 'filmListRow' : 'filmListColumn'}>
+                    {
+                        filmArr.map((el: FilmInterface, i: number) => {

```

Comment:
you don't need the index here, the filmInterface says the element has an id, use it as a key